src/link-manager.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/**
+ * @typedef {{
+ *   sourceViewName: string,
+ *   action: string,
+ *   targetViewName: string,
+ *   response: string
+ * }}
+ */
+genotet.LinkDef;
+
+/**
+ * @typedef {{
+ *   source: !genotet.View,
+ *   action: string,
+ *   target: !genotet.View,
+ *   response: string
+ * }}
+ */
+genotet.Link;
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Object that stores the link views by their type.
+ */
+/** @type {!Object<!Object<!Array<{
+ *    target: genotet.View,
+ *    response: string
+ *  }>>>}
+ */
+genotet.linkManager.links = {};
+
+/** @const {!Array<string>} */
+genotet.linkManager.EXPRESSION_ACTIONS = ['addProfile'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.NETWORK_ACTIONS = ['nodeClick', 'edgeClick'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.BINDING_ACTIONS = ['updateTrack', 'locus'];
+
+/** @const {string} */
+genotet.linkManager.DEFAULT_MAPPING = 'Direct Mapping';
+
+/**
+ * Initializes the link manager.
+ */
+genotet.linkManager.init = function() {
+  genotet.linkManager.links = {};
+  genotet.data.mappingFiles['gene-binding'] =
+    genotet.linkManager.DEFAULT_MAPPING;
+};
+
+/**
+ * Registers link action and response for all the views.
+ * @param {!Array<genotet.LinkDef>} links Link definition for all the views.
+ */
+genotet.linkManager.registerAllViews = function(links) {
+  links.forEach(function(link) {
+    var sourceView = genotet.viewManager.views[link.sourceViewName];
+    var targetView = genotet.viewManager.views[link.targetViewName];
+    /*
+     * TODO(Liana): Next version plans.
+     * Link settings will be extended to user self-defined in next version.
+     * We need check whether the view is existed, because link settings
+     * are defined as constant in this version, which cannot be revised
+     * corresponding to views.
+     */
+    genotet.linkManager.register_({
+      source: sourceView,
+      action: link.action,
+      target: targetView,
+      response: link.response
+    });
+  });
+};
+
+/**
+ * Registers link action and response for views.
```

Comment:
Registers one link. Pushes the link to adjacency list and sets up event listener.